### ISSUE TITLE
Cpp CodeQL  analysis fails (Exit code 99)

### ISSUE THREAD

**Issue by Nelson-numerical-software** [github/codeql#10194]
Cpp CodeQL  analysis fails
see original report here:
https://github.com/Nelson-numerical-software/nelson/issues/710

I tried to increase java heap size without success
see here: https://github.com/Nelson-numerical-software/nelson/runs/8048571951?check_suite_focus=true

**Comment by aibaars** [comment]
@Nelson-numerical-software could you try to revert to the previous version of CodeQL to test if there has been a performance regression somehow. The last working jobs ran with CodeQL 2.10.2 and the failures with CodeQL 2.10.3 . 

The easiest way to switch CodeQL version is to set the tag on the `codeql-action/init` step in your workflow to `v2.1.18`.

```
      # Initializes the CodeQL tools for scanning.
      - name: Initialize CodeQL
        uses: github/codeql-action/init@v2.1.18
```

See also: https://github.com/github/codeql-action/blob/a59fbe22086cdbbb45d9e16c202e29c781bf2d5e/CHANGELOG.md?plain=1#L23-L25

**Comment by aibaars** [comment]
Although on second thought using the `tools:` property of the `codeql-action/init` step may be better as the codeql-action may otherwise still pick up the default CodeQL version that is installed on the Actions VMs. The `tools` property needs to be set the the Linux artifact from https://github.com/github/codeql-action/releases/tag/codeql-bundle-20220728 :
```
      # Initializes the CodeQL tools for scanning.
      - name: Initialize CodeQL
        uses: github/codeql-action/init@v2
        with:
            tools: https://github.com/github/codeql-action/releases/download/codeql-bundle-20220728/codeql-bundle-linux64.tar.gz
            ...
```
If that work than you can even try the [pre-release](https://github.com/github/codeql-action/releases/tag/codeql-bundle-20220825) version to see if that version already fixed the performance problem. 

```
      # Initializes the CodeQL tools for scanning.
      - name: Initialize CodeQL
        uses: github/codeql-action/init@v2
        with:
            tools: https://github.com/github/codeql-action/releases/download/codeql-bundle-20220825/codeql-bundle-linux64.tar.gz
            ...
```

**Comment by Nelson-numerical-software** [comment]
Analysis fails again with v2.1.18
I modified .yml
https://github.com/Nelson-numerical-software/nelson/commit/8e9e8385926dd350cad38736b07b62d40c570d6b

see result:
https://github.com/Nelson-numerical-software/nelson/runs/8051871415?check_suite_focus=true

**Comment by aibaars** [comment]
@Nelson-numerical-software , that still ran with CodeQL [2.10.3](https://github.com/Nelson-numerical-software/nelson/runs/8051871415?check_suite_focus=true#step:3:16) unfortunately. Could you try the `tools:` property  as described in  https://github.com/github/codeql/issues/10194#issuecomment-1229219483?

**Comment by aibaars** [comment]
@Nelson-numerical-software I just found https://github.com/github/codeql/pull/10128 and https://github.com/github/codeql/pull/10117 . I looks like the team was already working on performance improvements for the CleartextBufferWrite.ql query. It is only partially fixed for 2.10.4. The fix relates to cases with integral types of which there are probably many in a numerical library. So fingers crossed that the partial fix is enough.

**Comment by aibaars** [comment]
@Nelson-numerical-software I'm closing this for now. We validated that the upcoming release (2.10.4) fixes the issue. Please re-open if you need further assistence.
